pkg/virtctl/ssh/native.go

```diff
@@ -0,0 +1,169 @@
+package ssh
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+
+	"github.com/golang/glog"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+	"golang.org/x/term"
+)
+
+func (o *SSH) prepareSSHClient(kind, namespace, name string) (*ssh.Client, error) {
+	streamer, err := o.prepareSSHTunnel(kind, namespace, name)
+	if err != nil {
+		return nil, err
+	}
+
+	conn := streamer.AsConn()
+	addr := fmt.Sprintf("%s/%s.%s", kind, name, namespace)
+	authMethods, err := o.getAuthMethods(kind, namespace, name)
+	if err != nil {
+		return nil, err
+	}
+
+	sshConn, chans, reqs, err := ssh.NewClientConn(conn,
+		addr,
+		&ssh.ClientConfig{
+			HostKeyCallback: ssh.InsecureIgnoreHostKey(),
```

Comment:
We should come up with a solution for this. Right now it is probably slightly less secure than `console` or `vnc`, since we directly connect from `virt-api` to the VMI IP, while for `vnc` and `console` we have a fully secured path over `virt-handler` to qemu sockets.

I think we can just read and write known_hosts files: https://pkg.go.dev/golang.org/x/crypto@v0.0.0-20210513164829-c07d793c2f9a/ssh/knownhosts#Line

Reply:
I added known_host checking and file writing.